### DISCUSSION TITLE
Prevent files from the root dir of the project being copied

### DISCUSSION THREAD

**Discussion by Steinkreis** [copier-org/copier#1448] [Q&A] [answered]
Is it possible to only copy the 📁 {{project_name}} and not the rest of the template? For example:

📁 my_copier_template                        # your template project
├── 📄 copier.yml                            # your template configuration
├── 📁 .git/                                 # your template is a Git repository
├── 📁 {{project_name}}                      # a folder with a templated name
│   └── 📄 {{module_name}}.py.jinja          # a file with a templated name
└── 📄 {{_copier_conf.answers_file}}.jinja  
└── 📄 pyproject.toml  

all files inside "my_copier_template" like the pyproject.toml will be copied to my cwd. This is something i don't want. I just want to get the project folder that i define with {{project_name}}. Is it possible to restrict this somehow?

**Comment by sisp** [comment]
I'm not sure I understand your problem. Why don't you want a file like `pyproject.toml` to be copied? Is this file part of the template and not part of the project to be generated?

**Reply by Steinkreis** [reply]
Hey thanks for your reply. I want to treat "my_copier_template" as my general template that i reference when i create new projects. Because "my_copier_template" is a repo itself with dependencies that need to be managed i use the pyproject.toml for that. However the actual template that i try to copy is the {{project_name}} folder. Inside that are several files like pre-defined pyproject.toml, pre-commit.yaml etc. When i run "copier copy my_copier_template ." I only want to copy the {{project_name}} folder and not the other files at the top level of "my_copier_template".

**Reply by sisp** [reply]
What you want is this filesystem layout

```diff
 📁 my_copier_template # your template project
 ├── 📄 copier.yml # your template configuration
 ├── 📁 .git/ # your template is a Git repository
-├── 📁 {{project_name}} # a folder with a templated name
+├── 📁 template/
+│   ├── 📄 {{_copier_conf.answers_file}}.jinja
 │   └── 📄 {{module_name}}.py.jinja # a file with a templated name
-├── 📄 {{_copier_conf.answers_file}}.jinja
 └── 📄 pyproject.toml
```

with `copier.yml` like this:

```yaml
_subdirectory: template
```

Unlike Cookiecutter, Copier doesn't use a templated directory name as the destination directory. Instead, you simply provide the destination directory via CLI, which is much more intuitive.

**Reply by Steinkreis** [reply]
Thanks again for replying. I tried it with you changes, however there are two problems now: 
- All the top level files are still being copied
- Now just the content of "template" is copied into my destination

My desired outcome would be however to do a `copier copy` to my template repo and create a new folder with the project_name inside my destination. 
So I would do  `copier copy my_copier_template . `. In my questions i would answer "new_project" on the project_name question. Finally i would get a new folder "new_project" in my cwd and no other files from the root of "my_copier_template".

**Reply by sisp** [reply, marked as answer]
This is not how Copier works. See also #1322. Deriving the destination directory's name from a question is not possible with Copier, and in fact it shouldn't be because (a) a directory with the same name might already exist and (b) the local directory name doesn't matter &ndash; it's only local on your machine anyway and doesn't affect the remote repo's name/path. If Copier worked like that, [applying multiple templates](https://copier.readthedocs.io/en/latest/configuring/#applying-multiple-templates-to-the-same-subproject) to the same destination directory wouldn't work.

Copier expects you to generate a project like this:

```shell
copier copy $src $dst
```

Whether `$dst` and `{{project_name}}` happen to be identical or not is your choice.

**Comment by pawamoy** [comment]
I think you're looking for [`_subdirectory: project_name`](https://copier.readthedocs.io/en/stable/configuring/#subdirectory) (its name shouldn't be templated)? However that will only render the Python module in the output directory. You probably want an additional directory layer.

**Reply by Steinkreis** [reply]
Hey thanks for your reply.
So basically my template repo looks like this

📁 my_copier_template # your template project
├── 📄 copier.yml # your template configuration
├── 📁 .git/ # your template is a Git repository
├── 📁 {{project_name}} # a folder with a templated name
│ └── 📄 {{module_name}}.py.jinja # a file with a templated name
│ └── 📄 other files
└── 📄 {{_copier_conf.answers_file}}.jinja
└── 📄 pyproject.toml

I want to "copier copy my_copier_template . " and only get following folders and files:

├── 📁 {{project_name}} # a folder with a templated name
│ └── 📄 {{module_name}}.py.jinja # a file with a templated name
│ └── 📄 other files


I tried to include `_subdirectory: "{{project_name}}"` in the copier.yaml but i get:
Traceback (most recent call last):
  File "~/.venv/bin/copier", line 8, in <module>
    sys.exit(copier_app_run())
  File "~/.venv/lib/python3.10/site-packages/plumbum/cli/application.py", line 638, in run
    inst, retcode = subapp.run(argv, exit=False)
  File "~/.venv/lib/python3.10/site-packages/plumbum/cli/application.py", line 633, in run
    retcode = inst.main(*tailargs)
  File "~/.venv/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "~/.venv/lib/python3.10/site-packages/copier/cli.py", line 72, in handle_exceptions
    return method(*args, **kwargs)
  File "~/.venv/lib/python3.10/site-packages/copier/cli.py", line 265, in main
    with self._worker(
  File "~/.venv/lib/python3.10/site-packages/copier/main.py", line 205, in __exit__
    raise value
  File "~/.venv/lib/python3.10/site-packages/copier/cli.py", line 272, in main
    worker.run_copy()
  File "~/.venv/lib/python3.10/site-packages/copier/main.py", line 754, in run_copy
    self._render_folder(src_abspath)
  File "~/.venv/lib/python3.10/site-packages/copier/main.py", line 643, in _render_folder
    for file in src_abspath.iterdir():
  File "/usr/lib/python3.10/pathlib.py", line 1017, in iterdir
    for name in self._accessor.listdir(self):
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/copier.vcs.clone.8c7rgngv/testtest '
